tests/integration_tests/performance/test_net_tput.py

```diff
@@ -0,0 +1,290 @@
+                results.append(result)
+                print(result)
+
+    return udp_statistics
+
+
+def run_tcp(microvm, host_ip, results, iperfs=1):
+    """Run tests using TCP protocol."""
+    protocol = 'tcp'
+    tcp_statistics = {}
+
+    for _ in range(RUNS):
+        for bw in BANDWIDTHS[protocol]:
+            for mode in MODES:
+                for packet_size in PACKET_SIZE:
+                    cmd = '{} -c {} -t{} -f MBytes -J -b {}M {} {} -w {}M \
+                            -O {} -l {} -P {} -A {}'.format(
+                        IPERF,
+                        host_ip,
+                        ITERATION_DURATION,
+                        bw,
+                        MODES_OPTIONS[mode],
+                        PROTOCOL_OPTIONS[protocol],
+                        WINDOW_SIZE,
+                        OMIT_TIME,
+                        packet_size,
+                        iperfs,
+                        PINNING_CPU
+                    )
+
+                    iperf_out = spawn_guest_iperf(microvm, cmd)
+                    result = parse_iperf_tcp_result(iperf_out, mode,
+                                                    bw, packet_size,
+                                                    tcp_statistics)
+                    results.append(result)
+                    print(result)
+
+    return tcp_statistics
+
+
+@pytest.mark.timeout(3600)
+@pytest.mark.env("benchmark")
+def test_network_tput_1_iperf(test_microvm_with_ssh, network_config):
+    """Execute network throughtput test."""
+    iperfs = 1
+    microvm = test_microvm_with_ssh
+    microvm.spawn()
+    microvm.basic_config(vcpu_count=iperfs + 1, mem_size_mib=1024)
+
+    spawn_host_iperf(microvm.jailer.netns_cmd_prefix())
+
+    _tap, host_ip, _ = microvm.ssh_network_config(network_config, '1')
+
+    microvm.start()
+    microvm.pin_vm_vcpus()
+
+    results = []
+    _ = run_udp(microvm, host_ip, results, iperfs)
+    _ = run_tcp(microvm, host_ip, results, iperfs)
+
+    # TODO : compute statistics
+
+    output_name = OUTPUT_JSON.format(iperfs)
+    with open(output_name, 'w') as outfile:
+        json.dump(results, outfile, indent=4)
+
+
+@pytest.mark.timeout(3600)
+@pytest.mark.env("benchmark")
+def test_network_tput_2_iperf(test_microvm_with_ssh, network_config):
+    """Execute network throughtput test."""
+    iperfs = 2
+    microvm = test_microvm_with_ssh
+    microvm.spawn()
+    microvm.basic_config(vcpu_count=iperfs + 1, mem_size_mib=1024)
+
+    spawn_host_iperf(microvm.jailer.netns_cmd_prefix())
+
+    _tap, host_ip, _ = microvm.ssh_network_config(network_config, '1')
+
+    microvm.start()
+    microvm.pin_vm_vcpus()
+
+    results = []
+    _ = run_udp(microvm, host_ip, results, iperfs)
+    _ = run_tcp(microvm, host_ip, results, iperfs)
+
+    # TODO : compute statistics
+
+    output_name = OUTPUT_JSON.format(iperfs)
+    with open(output_name, 'w') as outfile:
+        json.dump(results, outfile, indent=4)
```

Comment:
these 2 methods also look very similar. Can we have a method `test_network_tput(num_iperf_processes)` instead ? In order to have less duplicate code.

Reply:
for sure, will refactor this